src/wp-includes/update.php

```diff
@@ -956,6 +956,54 @@ function wp_clean_update_cache() {
 	delete_site_transient( 'update_core' );
 }
 
+/**
+ * Deletes all contents of the temp-backup directory.
+ *
+ * @since 6.0.0
+ *
+ * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+ */
+function wp_delete_all_temp_backups() {
+	/*
+	 * Check if there's a lock, or if currently performing an Ajax request,
+	 * in which case there's a chance we're doing an update.
+	 * Reschedule for an hour from now and exit early.
+	 */
+	if ( get_option( 'core_updater.lock' ) || get_option( 'auto_updater.lock' ) || wp_doing_ajax() ) {
+		wp_schedule_single_event( time() + HOUR_IN_SECONDS, 'wp_delete_temp_updater_backups' );
+		return;
+	}
+
+	add_action(
+		'shutdown',
+		/*
+		 * This action runs on shutdown to make sure there's no plugin updates currently running.
+		 * Using a closure in this case is OK since the action can be removed by removing the parent hook.
+		 * `remove_action( 'wp_delete_temp_updater_backups', 'wp_delete_all_temp_backups' );`
+		 */
+		static function() {
+			global $wp_filesystem;
+
+			if ( ! $wp_filesystem ) {
+				include_once ABSPATH . '/wp-admin/includes/file.php';
+				WP_Filesystem();
+			}
+
+			$temp_backup_dir = $wp_filesystem->wp_content_dir() . 'upgrade/temp-backup/';
```

Comment:
`$wp_filesystem->wp_content_dir()` returns `false` on failure or the location as a `string`. What should happen if it returns `false` meaning it failed to find the content directory? Should it throw an error and bail out?

Reply:
This question goes for all instances where `$wp_filesystem->wp_content_dir()` is used in this PR.

Reply:
I think it's probably best to throw the error and bail.